test/test-recipes.js

```diff
@@ -110,7 +110,100 @@ test('recipes with source', function(t) {
   });
   t.equal(g.definitionInterval, undefined, 'grammar definitionInterval is undefined');
   t.equal(g.ruleBodies.start.interval, undefined, 'ruleBodies.start.interval is undefined');
-  t.ok(toGrammar(g.toRecipe()), 'recipe still works fine');
+  t.ok(makeRecipe(g.toRecipe()), 'recipe still works fine');
 
   t.end();
 });
+
+test('semantics recipes', function(t) {
+  var g1 = ohm.grammar('G { Add = number "+" number  number = digit+ }');
+  var s1 = g1.semantics()
+    .addOperation('eval', {
+      Add: function(a, _, b) {
+        return a.value + b.value;
+      }
+    })
+    .addOperation('evalWith(inc, factor)', {
+      Add: function(a, _, b) {
+        return a.evalWith(this.args.inc, this.args.factor) +
+          b.evalWith(this.args.inc, this.args.factor);
+      },
+      number: function(digits) {
+        return (this.value + this.args.inc) * this.args.factor;
+      }
+    })
+    .addAttribute('value', {
+      number: function(digits) {
+        return parseFloat(this.interval.contents);
+      }
+    });
+
+  var s2 = makeRecipe(s1.toRecipe());
+  t.equal(s2.name, 'ASemantics', 'semantics created');
+
+  var g2 = s2._getSemantics().grammar; // FIXME: should have an official API
```

Comment:
I suppose adding a `getGrammar` method to the `Semantics` wrappers would be OK. But what I think we really want is an object that represents an "ohm bundle" that would have a grammar and a semantics, as well as examples and tests to go with those. (These examples and tests should probably be included in their associated grammar and semantics, instead of being part of the "bundle".) The three of us should talk about this soon.